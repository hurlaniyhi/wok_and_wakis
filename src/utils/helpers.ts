import { FoodItem } from "@/types"

const sortFoodItems = (category: string, items: Array<FoodItem>) => {
    const grouped: Record<string, FoodItem[]> = {};
  
    const addToGroup = (key: string, item: FoodItem) => {
        if (!grouped[key]) {
            grouped[key] = [item];
        } else {
            grouped[key].push(item);
        }
    };
  
    for (const item of items) {
        if (category === 'Popular Items') {
            addToGroup(item.category, item);
    
            if (item.isPopular) {
                addToGroup('Popular Items', item);
            }
        } else if (item.category === category) {
            addToGroup(item.category, item);
        }
    }
  
    // Ensure "Popular Items" is first if it exists
    if ('Popular Items' in grouped) {
        const { ['Popular Items']: popularItems, ...rest } = grouped;
        
        return {
            'Popular Items': popularItems,
            ...rest,
        };
    }
  
    return grouped;
};

const wait = (time: number) => new Promise(res => setTimeout(res, time))

export default {
    sortFoodItems,
    wait
}