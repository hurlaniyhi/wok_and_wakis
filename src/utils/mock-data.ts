import { FoodItem } from "@/types";

const foodItems: Array<FoodItem> = [
    {
        "id": 1,
        "title": "Hat Trick Breakfast",
        "description": "Three eggs any style, three strips of bacon or three sausage links and this is very good and okay",
        "price": 18.69,
        "category": "Sandwiches",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/FHwht2jp/three.jpg"
    },
    {
        "id": 2,
        "title": "Double Down Breakfast",
        "description": "Two eggs any style, two strips of bacon or two sausage links, toast",
        "price": 19.85,
        "category": "Sandwiches",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/nck6YNRr/double.jpg"
    },
    {
        "id": 3,
        "title": "Egg Bagel Sandwich",
        "description": "Fried egg, American chees and diced ham, baconn strips or a sausage patty and this is very good and okay",
        "price": 15.75,
        "category": "Sandwiches",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/28b03Jsg/casey.jpg"
    },
    {
        "id": 4,
        "title": "Wings & fries",
        "description": "Comes with Fries",
        "price": 18.46,
        "category": "Wings & Tenders",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/zD9cw8Fq/wings.jpg"
    },
    {
        "id": 5,
        "title": "Fish Tray",
        "description": "Comes with 6 free Sauces",
        "price": 18.00,
        "category": "Party Trays",
        "isPopular": false,
        "imageUrl": "https://i.postimg.cc/kXmk376f/fish.jpg"
    },
    {
        "id": 6,
        "title": "Pick 2 & Fries",
        "description": "Comes with fries and any sauce you want",
        "price": 8.50,
        "category": "Mix & Match Combos",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/0yZn4sDM/edgar.jpg"
    },
    {
        "id": 7,
        "title": "Hush Puppies",
        "description": "",
        "price": 13.97,
        "category": "Mix & Match Combos",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/rwzR8X7t/cala.jpg"
    },
    {
        "id": 8,
        "title": "Mozarella Sticks",
        "description": "Comes with Marinara Sauce",
        "price": 14.09,
        "category": "Kidâ€™s Menu",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/fLHhWFGJ/stick.jpg"
    },
    {
        "id": 9,
        "title": "Shrimp & Fries",
        "description": "Comes with Fries",
        "price": 5.02,
        "category": "Fish & Shrimp",
        "isPopular": true,
        "imageUrl": "https://i.postimg.cc/KcM2h99v/shrimp.jpg"
    }, 
    {
        "id": 10,
        "title": "Tender Tray",
        "description": "Comes with 6 free Sauces",
        "price": 35.00,
        "category": "Party Trays",
        "isPopular": false,
        "imageUrl": "https://i.postimg.cc/0QSHF49T/tender.jpg"
    },
    {
        "id": 11,
        "title": "WIngs Tray",
        "description": "Comes with 6 free Sauces",
        "price": 22.00,
        "category": "Party Trays",
        "isPopular": false,
        "imageUrl": "https://i.postimg.cc/pTQJDMWL/ros.jpg"
    }
]

const cartItems = [
    {
        "selectedCheeses": [
            "Add Swiss",
            "Add Pepper Jack",
            "Add Provolone"
        ],
        "eggStyle": "Over Hard",
        "instructions": {
            "kitchenNote": "",
            "soldOutOption": ""
        },
        "quantity": 1,
        "item": {
            "id": 1,
            "title": "Pick 2 & Fries",
            "description": "Comes with fries and any sauce you want",
            "category": "Mix & Match Combos",
            "price": 8.5,
            "isPopular": true,
            "imageUrl": "https://i.postimg.cc/0yZn4sDM/edgar.jpg"
        },
        "cartId": 1
    }
]

export default {
    foodItems,
    cartItems
}