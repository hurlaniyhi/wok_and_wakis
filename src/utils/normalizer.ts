const requestMessage = (resp: any, text = '') => {
    return resp?.message || text || 'Something went wrong'
}

const catchErrMsg = (err: any) => {
    return (
        err?.response?.data?.errors?.[0]?.message ||
        err?.response?.data?.message ||
        'Something went wrong, please try again later.'
    );
}

const isSuccessResponse = (data: any, status: number) => {
    return [200, 201].includes(status) && data?.status === 'success';
}

export {
    requestMessage,
    catchErrMsg,
    isSuccessResponse
}