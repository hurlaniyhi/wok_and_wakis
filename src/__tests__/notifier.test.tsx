import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Notifier  from "@/components/feedback/notifier"
import { TestProviders } from '@/__tests__/utils/test-provider'

jest.mock('../providers/context', () => {
    const actual = jest.requireActual('../providers/context');
    return {
        ...actual,
        useFeedbackContext: () => ({
            feedback: {
                notification: {
                    title: '',
                    message: 'Test message',
                    status: true,
                }
            },
            setFeedback: jest.fn(),
        }),
    };
});
 
describe('Notifier', () => {
    it('renders the notifier (toaster)', () => {
        render(<Notifier />, { wrapper: TestProviders })
        expect(screen.getByTestId("notifier")).toBeInTheDocument();
    });

    it('displays a toast message', () => {
        render(<Notifier />, { wrapper: TestProviders });
        expect(screen.getByText(/test message/i)).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Notifier />, { wrapper: TestProviders });
        expect(asFragment()).toMatchSnapshot();
    });
})