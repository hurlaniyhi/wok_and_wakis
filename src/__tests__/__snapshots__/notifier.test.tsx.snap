// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Notifier matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-904fb7c6-16 bRGBqy"
    data-testid="notifier"
  >
    <div
      class="sc-904fb7c6-2 frLkSI"
    >
      <div
        class="sc-904fb7c6-17 ktMLfa fixed_flex_shape"
        overflow="hidden"
        size="4.5"
      >
        <svg
          color="#B42318"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(180, 35, 24);"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 0 0-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
          />
          <path
            d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </div>
      <div
        class="sc-904fb7c6-0 uuTFX"
        width="27"
      >
        <p
          class="sc-904fb7c6-5 edFMAg"
          color="#667085"
        >
          Test message
        </p>
        <span
          class="sc-904fb7c6-6 jgAJLw"
          color="#B42318"
          cursor="pointer"
          font-weight="600"
        >
          Dismiss
        </span>
      </div>
      <button
        class="sc-904fb7c6-4 fewQtc"
        color="#000000"
        font-weight="500"
        height="3.4"
        radius="0.4"
        width="3.4"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="margin-top: 0.5rem;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
